### PR TITLE
270 Add Base for the databse migrations

### PR DESCRIPTION
**Pull request by maierdominik**
New start without using an external plugin

270: 
* adapt schema sucht that each document has the property `schemaVersion`
* create migration where the value for `schemaVersion` is set to `0` to test it

### CODE REVIEW

File: apps/frontend/src/pages/ClassManagement/components/GroupDialog/GroupPropertiesTable.tsx [hunk 1]
```diff
@@ -89,7 +89,7 @@ const GroupPropertiesTable = ({ isCreateMode, disabled, form }: GroupPropertiesT
       case 'text':
       default:
         if (groupProperty.disabled) {
-          return <>{watch(groupProperty.name)}</>;
+          return watch(groupProperty.name) as string;
```
Comment (TomlDev) [inline apps/frontend/src/pages/ClassManagement/components/GroupDialog/GroupPropertiesTable.tsx:92]: Instead change line 14 to 
`name: keyof Omit<GroupForm, 'admins' | 'admingroups' | 'members' | 'membergroups'>`

File: apps/api/src/appconfig/appconfig.migration.service.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @InjectConnection() private readonly connection: Connection,
+    @InjectModel(AppConfig.name) private readonly appConfigModel: Model<AppConfig>,
+  ) {}
+
+  async onModuleInit() {
+    await initializeCollection(this.connection, this.appConfigModel);
+
+    Logger.log(`⬆ ️⬆️ Migrating`, AppConfigMigrationService.name);
+
+    Logger.log('⬆ ... executing Migrations', AppConfigMigrationService.name);
```
Comment (TomlDev) [inline apps/api/src/appconfig/appconfig.migration.service.ts:19]: I would combine the two logging lines
```
    Logger.log(`⬆ ️⬆️ Executing migratings`, AppConfigMigrationService.name);

File: apps/api/src/appconfig/appconfig.migration.service.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    // eslint-disable-next-line no-underscore-dangle
+    const ids = unprocessedDocuments.map((d) => d._id);
+
+    try {
+      await this.appConfigModel.updateMany(
+        { _id: { $in: ids } },
+        {
+          $set: {
+            migrationNumber: newMigrationNumber,
```
Comment (TomlDev) [inline apps/api/src/appconfig/appconfig.migration.service.ts:49]: I dislike the name `migrationNumber`, what about `schemaVersion`, this is also used in the MongoDB docs

File: apps/api/src/appconfig/appconfig.migration.service.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            updatedAt: new Date().toDateString(),
+          },
+        },
+      );
+      Logger.log(`⬆ Migration "${migrationName}" was successfully completed.`, AppConfigMigrationService.name);
+    } catch (e) {
+      Logger.error(`⬆     Error while running Migration "${migrationName}": ${e}`, AppConfigMigrationService.name);
```
Comment (TomlDev) [inline apps/api/src/appconfig/appconfig.migration.service.ts:56]: Is there a reason for the whitespaces?

**Review by TomlDev — CHANGES_REQUESTED** [review]
Looks good 👍🏽

File: apps/api/src/appconfig/appconfig.migration.service.ts (hunk 4 above)
Comment (maierdominik) [reply apps/api/src/appconfig/appconfig.migration.service.ts:56]: Reverted.
My thought were that it is easier to differ this line of output from all the other lines.

**Review by TomlDev — APPROVED** [review]
👍🏽

**Pull request merged**